dotnet/src/Connectors/Connectors.Memory.SqlServer/ISqlServerClient.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel.Connectors.SqlServer;
+
+/// <summary>
+/// Interface for client managing SQL Server or Azure SQL database operations.
+/// </summary>
+public interface ISqlServerClient
```

Comment:
`ISqlServerClient` is public because it can be passed into `SqlServerMemoryStore`s ctor. Thus also `SqlServerMemoryEntry` is public. `SqlServerClient` can be made internal.

If we would remove the ctor overload from `SqlServerMemoryStore` then everything can be made internal. Not sure whether we want to do it.

Reply:
@cincuranet, I'd recommend making the constructor internal too. It's easy to make it public if someone wants to replace just that piece. Until we have such a request though it's better not to have to support that publicly.

Reply:
OK.